derp: TestSendFreeze is flaky

Recently flaked on CI. Reproducible locally with `go test -c -race . && stress ./derp.test`. That averages one failure every 500 runs or so. Sample failure:

```
--- FAIL: TestSendFreeze (0.23s)
    logger.go:31: derp client alice/f8546d7fdecfc51f9b7b932646e2df83540147cfcf5f95bff0e5e9aefc90ba7a: adding connection
    logger.go:31: derp client bob/fea05b3fa62641f74e1f6af6942740ff46b84a5d279460f37c5971fc2e559f10: adding connection
    logger.go:31: derp client cathy/2f0a17cf76a1aa5d20c1d6ac1c3a9e4ec5ee000f2fad416ab0bc7a0d6e91f205: adding connection
    derp_test.go:403: TEST COMPLETE, cancelling sender
    derp_test.go:405: closing connections
    logger.go:31: derp client alice/f8546d7fdecfc51f9b7b932646e2df83540147cfcf5f95bff0e5e9aefc90ba7a: read EOF
    logger.go:31: derp client alice/f8546d7fdecfc51f9b7b932646e2df83540147cfcf5f95bff0e5e9aefc90ba7a: removing connection
    derp_test.go:416: bob: unexpected message type derp.PeerGoneMessage
    logger.go:31: derp client cathy/2f0a17cf76a1aa5d20c1d6ac1c3a9e4ec5ee000f2fad416ab0bc7a0d6e91f205: read EOF
    logger.go:31: derp client bob/fea05b3fa62641f74e1f6af6942740ff46b84a5d279460f37c5971fc2e559f10: read EOF
    logger.go:31: derp client bob/fea05b3fa62641f74e1f6af6942740ff46b84a5d279460f37c5971fc2e559f10: removing connection
    logger.go:31: derp client cathy/2f0a17cf76a1aa5d20c1d6ac1c3a9e4ec5ee000f2fad416ab0bc7a0d6e91f205: removing connection
FAIL
```

I went to bisect, but the test appears to have been flaky from the beginning.

cc @crawshaw

--- a/derp/derp_test.go
+++ b/derp/derp_test.go
@@ -403,8 +403,11 @@ func TestSendFreeze(t *testing.T) {
 	t.Logf("TEST COMPLETE, cancelling sender")
 	cancel()
 	t.Logf("closing connections")
-	aliceConn.Close()
+	// Close bob before alice.
+	// Starting with alice can cause a PeerGoneMessage to reach
+	// bob before bob is closed, causing a test flake (issue 2668).
 	bobConn.Close()
+	aliceConn.Close()
 	cathyConn.Close()
 
 	for i := 0; i < cap(errCh); i++ {

